packages/lib/SyncTargetJoplinCloud.ts

```diff
@@ -54,9 +58,29 @@ export default class SyncTargetJoplinCloud extends BaseSyncTarget {
 			const fileApi = await this.fileApi();
 			const api = fileApi.driver().api();
 			const sessionId = await api.sessionId();
-			return !!sessionId;
+			if (!sessionId) {
+				return false;
+			}
+
+			const settings = Setting.toPlainObject();
+			const options = {
+				...Setting.subValues(`sync.${SyncTargetJoplinCloud.id()}`, settings),
+				...Setting.subValues('net', settings),
+			};
+
+			const result = await SyncTargetJoplinCloud.checkConfig(convertValuesToFunctions(options));
+
+			if (result.errorMessage) {
+				logger.error(result.errorMessage);
+			}
+
+			Setting.setValue(`sync.${SyncTargetJoplinCloud.id()}.isAuthenticated`, result.ok);
+			return result.ok;
 		} catch (error) {
-			if (error.code === 403) return false;
+			if (error.code < 500) {
+				Setting.setValue(`sync.${SyncTargetJoplinCloud.id()}.isAuthenticated`, false);
+				return false;
+			}
```

Comment:
I'm not sure we should have this logic here because when is it actually called? Also it's called "isAuthenticated" but it's clearly doing a lot more than this now, including saving settings.

Is it possible to add the code to lib/Synchronizer.ts instead, with the same logic as "MustUpgradeApp" error? If we catch an error, and if it's an auth error, and it's Joplin Cloud, then dispatch an action telling the user they must authenticate. That way we deal with the problem exactly where it happens.

Reply:
I'm going to still test your suggestion, and I see how what I'm doing is really not necessary at this point (I'm removing the `Setting.setValue` and `checkConfig`), but I think we still need to have some error catch logic here.

It is necessary because when we call `api.sessionId()` and the user has invalid credentials it will throw an error (probably 403 or 404, depending on why the credentials are invalid).

The credentials can be invalid if the user:
- Has enabled MFA and hasn't made a new login in the client yet.
- Has deleted the application record from the Joplin Cloud page.